core/platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/SingleEnvironmentFilter.java

```diff
@@ -101,8 +71,10 @@ private void setWorkspaceIdToContext(EnvironmentContext environmentContext, Http
 
         String[] uriParts = requestURI.split("/", 5);
 
-        if (requestURI.contains("/ws") && uriParts.length >= 4) {
-            environmentContext.setWorkspaceId(uriParts[4]);
+        for (String part : uriParts) {
```

Comment:
we have to get workspace id from URI. Workspace id starts with 'workspace'. Maybe uriParts is not  a good name.